sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -336,12 +338,15 @@ private Mono<Void> refreshLocation(boolean isReadRequest, boolean forceRefresh,
         this.failoverRetryCount++;
 
         // Mark the current read endpoint as unavailable
+
+        URI gatewayRegionalEndpoint = this.regionalRoutingContext.getGatewayRegionalEndpoint();
+
         if (isReadRequest) {
-            logger.warn("marking the endpoint {} as unavailable for read",this.locationEndpoint);
-            this.globalEndpointManager.markEndpointUnavailableForRead(this.locationEndpoint);
+            logger.warn("marking the endpoint {} as unavailable for read", gatewayRegionalEndpoint);
```

Comment:
@jeet1995 - NIT: with PPAF is the markdown here based on endpoint or region? I guess we would make the decision always for region (and not have gateway endpoint marked down, but thin client marked as healthy). We can probably reflect that always as gateway endpoint and then just have an in-region state that tells whether thin clinet endpoint is healthy or we should skip to gateway?

Reply:
You're right - PPAF mark down conceptually is per region per partition and not `URI`. PPAF will still accept `RegionalRoutingContext` as a contract but implementation of mark down should be geared towards region.